File: backend/internal/models/usuario.go
package models

type Usuario struct {
	ID         int      `json:"id"`
	Nome       string   `json:"nome"`
	Username   string   `json:"username"`
	SenhaHash  string   `json:"-"`
	LojaID     int      `json:"loja_id"`
	LojaNome   string   `json:"loja_nome,omitempty"`
	Perfil     string   `json:"perfil"`
	Permissoes []string `json:"permissoes"`
	Ativo      bool     `json:"ativo"`
	CriadoEm   string   `json:"criado_em"`
}

File: frontend/src/types/Usuario.ts
export interface Usuario {
  id?: number;
  nome: string;
  username: string;
  senha_hash?: string;
  loja_id: number;
  perfil: "admin" | "operador" | "visualizador" | string;
  permissoes: string[];
  ativo?: boolean;
  criado_em?: string;
}

export interface UsuarioPayload {
  nome: string;
  username: string;
  senha: string;
  loja_id: number;
  perfil: string;
  permissoes: string[];
}
